include/dragonbox/dragonbox.h

```diff
@@ -114,17 +148,25 @@ namespace jkj::dragonbox {
         // Depending on the floating-point encoding format, this operation might not be possible for
         // some specific bit patterns. However, the contract is that u always denotes a
         // valid bit pattern, so this function must be assumed to be noexcept.
-        static T carrier_to_float(carrier_uint u) noexcept {
+        static JKJ_CONSTEXPR20 T carrier_to_float(carrier_uint u) noexcept {
+#if JKJ_HAS_BIT_CAST
+            return std::bit_cast<T>(u);
+#else
             T x;
             std::memcpy(&x, &u, sizeof(carrier_uint));
             return x;
+#endif
         }
 
         // Same as above.
-        static carrier_uint float_to_carrier(T x) noexcept {
+        static JKJ_CONSTEXPR20 carrier_uint float_to_carrier(T x) noexcept {
+#if JKJ_HAS_BIT_CAST
+            return std::bit_cast<carrier_uint>(x);
+#else
             carrier_uint u;
             std::memcpy(&u, &x, sizeof(carrier_uint));
             return u;
+#endif
         }
```

Comment:
I think it would be better to have a separate free function `bit_cast` in the `detail` namespace which calls `std::bit_cast` and falls back to `std::memcpy` if it's not possible.